packages/menu/src/useMenu.ts

```diff
@@ -12,7 +12,6 @@ import React, {
   useEffect,
```

Comment:
I assume there are already some tests capturing this functionality well enough that we don't need to update/add more tests?

Reply:
Yes, there are numerous tests in `MenuContainer.spec.tsx` that check for the correct focused element. However, none of them track the order of execution. Would you like me to add tests asserting that the focus is returned before the `onChange` callback is fired?

Reply:
Oh, no I think it's fine as-is. :)